app/workers/delete_plain_object_worker.rb

```diff
@@ -34,8 +34,8 @@ class DeletePlainObjectWorker < ActiveJob::Base
     info "Finished #{job.class}#perform with the hierarchy of workers: #{caller_worker_hierarchy}"
   end
 
-  def perform(_object, _caller_worker_hierarchy = [])
-    should_destroy_by_association? ? destroy_by_association : object.destroy!
+  def perform(_object, _caller_worker_hierarchy = [], _destroy_method = 'destroy')
+    should_destroy_by_association? ? destroy_by_association : object.public_send("#{destroy_method}!")
```

Comment:
Does `delete!` exist? :thinking:

Reply:
gosh, you're right. delete does not have a bang version.. need to write a test for that.